bc_obps/compliance/service/elicensing/elicensing_obligation_service.py

```diff
@@ -129,7 +139,13 @@ def process_obligation_integration(cls, obligation_id: int) -> None:
 
                 # If successful, update the compliance status
                 ComplianceReportVersionService.update_compliance_status(obligation.compliance_report_version)
-                retryable_send_notice_of_obligation_due_email.execute(obligation.id)
+                # If the new obligation was created by a supplementary report and the obligation deadline has passed, we send a specific email. Otherwise, we send the general one about the obligation being due.
+                if obligation.compliance_report_version.is_supplementary and cls._has_obligation_deadline_passed(
+                    obligation.obligation_deadline
+                ):
+                    retryable_notice_of_supplementary_report_post_deadline_increases_emissions.execute(obligation.id)
```

Comment:
Since this is related to a supplementary report, should we call this task somewhere in the `supplementary_version_service.py` file instead?

Reply:
I'm going to answer this question with a question! I put the email logic in `process_obligation_integration` because I wanted to be sure that we only sent the email if the invoice creation and data sync succeeded. If I were to move the email sending function to the supp report service and call it after `ElicensingObligationService.handle_obligation_integration(obligation.id, compliance_report.compliance_period)`, would a failure in `handle_obligation_obligation_integration` always prevent sending of the email? The internet is telling me conflicting things about how errors in retryables work.